### PR TITLE
chore: add caption to user links

### PR DESCRIPTION
**Pull request by gtoprak**
Created a shared component `Showcase` and updated the usages around the site. Effects `index` and `users`. 

## Motivation

Fixes https://github.com/facebook/Docusaurus/issues/1040.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Clean linter, and visited pages affected.

## Related PRs

There was already an open PR https://github.com/facebook/Docusaurus/pull/1043 by @mptap, but I've taken a significantly different approach and trying to remove some repetition of code.

### CODE REVIEW

**Comment by gtoprak** [comment]
@yangshun — Can you give some details about why that's the case? Is that something related to caching? If that's the case do I have to change other parts on the PR too?

**Comment by yangshun** [comment]
@gtoprak I just realized we met at React Conf! Thanks for your contribution!

Hmm it's because `lib/static/main.css` is part of the `docusaurus` library. This showcase thing is generated for all projects when someone does `docusaurus init` and we won't have control over those code after that. If we remove the `productShowcaseSection`, existing users' HTML will still have the class but there's no longer any styling associated with the class. I recommend not changing any CSS classes. We can only add more classes, but even that is not ideal because *ALL* users will get the new CSS even though they're using it. Ideally we just add HTML and the custom HTML we add to `v1/website/static/css/custom.css`.

**Comment by gtoprak** [comment]
@yangshun — Yes, thank you for the sticker! I am hoping to have many more contributions in the future too.

I've made the updates to `custom.css` and reverted my changes to `main.css`. As far as I can tell from the _netlify_ deploy, it seems to be working as it does on my local. But, please do not hesitate to let me know if this work still is not complete, as you have the expertise on how Docusaurus functions for the users, and I'll lean on you on that.

File: v1/lib/core/CompLibrary.js [hunk 1]
```diff
@@ -8,10 +8,12 @@
 const MarkdownBlock = require('./MarkdownBlock.js');
 const Container = require('./Container.js');
 const GridBlock = require('./GridBlock.js');
+const Showcase = require('./Showcase.js');
```
Comment (yangshun) [inline v1/lib/core/CompLibrary.js]: Don't make it part of the exported component library. The custom styles are not part of the library and users who import the library will see unstyled contents.

File: v1/lib/core/Showcase.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const PropTypes = require('prop-types');
+
+class UserLink extends React.Component {
```
Comment (yangshun) [inline v1/lib/core/Showcase.js]: Make this a functional component.

File: v1/lib/core/Showcase.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  }
+}
+
+UserLink.propTypes = {
+  infoLink: PropTypes.string.isRequired,
+  image: PropTypes.string.isRequired,
+  caption: PropTypes.string.isRequired,
+};
+
+class Showcase extends React.Component {
```
Comment (yangshun) [inline v1/lib/core/Showcase.js]: Functional component.

File: v1/lib/static/css/main.css [hunk 4]
```diff
@@ -2353,4 +2353,4 @@ input::placeholder {
 .nav-footer .social {
   padding: 5px 0;
 }
-/* End of Footer */
+/* End of Footer */
```
Comment (yangshun) [inline v1/lib/static/css/main.css]: There's a trailing newline here which got removed.

File: v1/website/pages/en/users.js [hunk 5]
```diff
@@ -9,26 +9,27 @@ const React = require('react');
...
 const translate = require('../../server/translate.js').translate;
 
 class Users extends React.Component {
-  renderUser(user) {
-    return (
-      <a href={user.infoLink} key={user.infoLink}>
-        <img src={user.image} alt={user.caption} title={user.caption} />
-      </a>
-    );
+  getUsersToShowcase() {
```
Comment (yangshun) [inline v1/website/pages/en/users.js]: The original way, although it needs 2 passes of the array, is much more declarative. Please revert to that approach.

**Comment by gtoprak** [comment]
I've made the suggested updates, and I did have an off alignment on certain user images. Following two screenshots are how it's looking for me, please let me know if that doesn't reflect what the team is envisioning.

@yangshun — Thank you for the ongoing code review as I appreciate it a lot since I am not familiar with the code base, it's really helpful!

screen shot 2018-10-30 at 5 19 13 pm
screen shot 2018-10-30 at 5 19 34 pm

**Comment by gtoprak** [comment]
Thank you @yangshun!

**Pull request merged**